### PR TITLE
Use stock grafana images instead of custom docker build

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR moves Istio off of the process of building our own custom grafana image that loads the Istio dashboards and sets up the Prometheus datasource. It takes advantage of the provisioning features that were introduced in grafana 5.x.x. This should slightly improve overall build times and reduce user confusion over custom docker images.

It achieves this transition by creating two new configmaps to hold provisioning config and the dashboards.

Reviewers: Please please please try this out and confirm that it works for you as well.

### CODE REVIEW

**Comment by jeffmendoza** [comment]
Looks good, I'll try out generating the yamls from the helm templates and deploying grafana.

**Comment by douglas-reid** [comment]
@jeffmendoza i've never been able to successfully run helm just on any specific template. If you want to generate with grafana, you have to add `--set grafana.enabled=true` or whatever in your helm command.

File: install/kubernetes/helm/istio/charts/grafana/templates/configmap-dashboards.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "grafana.fullname" . }}-dashboards
```
Comment (jeffmendoza) [inline install/kubernetes/helm/istio/charts/grafana/templates/configmap-dashboards.yaml]: Need namespace here.

File: install/kubernetes/helm/istio/charts/grafana/templates/configmap.yaml [hunk 2]
```diff
@@ -1,16 +1,24 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: istio-grafana-custom-resources
-  namespace: {{ .Release.Namespace }}
+  name: {{ template "grafana.fullname" . }}
```
Comment (jeffmendoza) [inline install/kubernetes/helm/istio/charts/grafana/templates/configmap.yaml]: Need namespace here too.

File: install/kubernetes/helm/istio/charts/grafana/templates/configmap-dashboards.yaml (hunk 1 above)
Comment (douglas-reid) [reply install/kubernetes/helm/istio/charts/grafana/templates/configmap-dashboards.yaml]: Added.

**Comment by jeffmendoza** [comment]
I'm seeing the columns not ligning up with the data in the mesh dashboard:
mesh dashboard
I don't see how that could be caused by this PR. I'll check HEAD. I was looking at the 1.0 release before and it looked right.

**Comment by douglas-reid** [comment]
@jeffmendoza any luck with figuring out the dashboard alignment issue?

@sdake any updates?

**Comment by sdake** [comment]
@douglas-reid i'll test now, but lgtm.

File: install/kubernetes/helm/istio/charts/grafana/templates/deployment.yaml [hunk 3]
```diff
@@ -78,13 +74,27 @@ spec:
...
+            subPath: datasources.yaml
+          - name: config
+            mountPath: "/etc/grafana/provisioning/dashboards/dashboardproviders.yaml"
+            subPath: dashboardproviders.yaml
       affinity:
       {{- include "nodeaffinity" . | indent 6 }}
       volumes:
+      - name: config
+        configMap:
+          name: {{ template "grafana.fullname" . }}
```
Comment (sdake) [inline install/kubernetes/helm/istio/charts/grafana/templates/deployment.yaml]: just a general comment - I believe we had removed all the templates for `fullname` but either that wasn't the case or they snuck back into the codebase. 
```
sdake@falkor-07:~/go/src/istio.io/istio/install/kubernetes/helm/istio/charts$ grep -r template * | grep full
grafana/templates/deployment.yaml:          name: {{ template "grafana.fullname" . }}
grafana/templates/deployment.yaml:          name:  {{ template "grafana.fullname" . }}-dashboards
grafana/templates/configmap-dashboards.yaml:  name: {{ template "grafana.fullname" . }}-dashboards
grafana/templates/configmap.yaml:  name: {{ template "grafana.fullname" . }}
servicegraph/templates/ingress.yaml:  name: {{ template "servicegraph.fullname" . }}
servicegraph/templates/NOTES.txt:  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "servicegraph.fullname" . }})
servicegraph/templates/NOTES.txt:           You can watch the status of by running 'kubectl get svc -w {{ template "servicegraph.fullname" . }}'
servicegraph/templates/NOTES.txt:  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "servicegraph.fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
tracing/templates/ingress.yaml:  name: {{ template "zipkin.fullname" . }}
```

**Comment by douglas-reid** [comment]
@sdake @jeffmendoza OK. Updated to fix names and strange dashboard drift. PTAL.

**Comment by douglas-reid** [comment]
@sdake can i have another '/lgtm' ? I've updated the tag from `5.2.2` to `5.2.3`, and that's it.

**Comment by sdake** [comment]
@douglas-reid can you revert your last version change to determine if that is causing the gate implosion on circleci and prow?

**Comment by douglas-reid** [comment]
@sdake gate implosion?

**Comment by sdake** [comment]
@douglas-reid just a comment on the red gates..

/retest

**Comment by sdake** [comment]
@douglas-reid it appears to me that the v1alpha3.sh tests either do not wait a sufficient amount of time for the bookinfo pods to start up, or there is a capacity problem on the system under test, or there is a legitimate bug in master.

```
NAME                                        READY     STATUS            RESTARTS   AGE
details-v1-54c8684f-mmqcg                   2/2       Running           0          8m
details-v2-5cfcf65fc8-c6sfh                 0/2       PodInitializing   0          8m
istio-citadel-9d98bb886-mn226               1/1       Running           0          9m
istio-egressgateway-577cc4f8c4-szp6v        1/1       Running           0          9m
istio-galley-579cf7b44d-vpwq8               1/1       Running           0          9m
istio-ingressgateway-86b7c9fbfc-498wl       1/1       Running           0          9m
istio-pilot-6759fb97c6-rm7nw                2/2       Running           0          9m
istio-policy-c64cdbfc6-k7l77                2/2       Running           0          9m
istio-sidecar-injector-54977b95b6-9r29j     1/1       Running           0          9m
istio-statsd-prom-bridge-55965ff9c8-v9ddm   1/1       Running           0          9m
istio-telemetry-6585c6f785-k7gk4            2/2       Running           0          9m
mongodb-v1-7f9844b9bf-rqfgs                 0/2       PodInitializing   0          8m
mysqldb-v1-75df7bf7d7-mwsvw                 2/2       Running           0          8m
productpage-v1-778576d5c-dgw52              2/2       Running           0          8m
prometheus-7456f56c96-nw4gs                 1/1       Running           0          9m
ratings-v1-5794cf87b9-6pm7h                 2/2       Running           0          8m
ratings-v2-56b77b55f8-6h6lx                 0/2       PodInitializing   0          8m
ratings-v2-mysql-6d6597f657-kxrsp           2/2       Running           0          8m
reviews-v1-9f8b59bcf-h72n5                  2/2       Running           0          8m
reviews-v2-8658564757-qchzr                 0/2       PodInitializing   0          8m
reviews-v3-7bfb794f5f-88q8z                 2/2       Running           0          8m
details istio-proxy
details istio-proxy
citadel
istio-proxy
galley
istio-proxy
discovery istio-proxy
mixer istio-proxy
sidecar-injector-webhook
statsd-prom-bridge
mixer istio-proxy
mongodb istio-proxy
mysqldb istio-proxy
productpage istio-proxy
prometheus
ratings istio-proxy
ratings istio-proxy
ratings istio-proxy
reviews istio-proxy
reviews istio-proxy
reviews istio-proxy




* command failed: "Error from server (BadRequest): container \"details\" in pod \"details-v2-5cfcf65fc8-c6sfh\" is waiting to start: PodInitializing\n" exit status 1
* command failed: "Error from server (BadRequest): container \"mongodb\" in pod \"mongodb-v1-7f9844b9bf-rqfgs\" is waiting to start: PodInitializing\n" exit status 1
* command failed: "Error from server (BadRequest): container \"ratings\" in pod \"ratings-v2-56b77b55f8-6h6lx\" is waiting to start: PodInitializing\n" exit status 1
* command failed: "Error from server (BadRequest): container \"reviews\" in pod \"reviews-v2-8658564757-qchzr\" is waiting to start: PodInitializing\n" exit status 1

```

**Comment by douglas-reid** [comment]
Can we merge this?  All required tests are passing.

**Comment by hklai** [comment]
/lgtm

@douglas-reid do you anticipate any impact to release?

**Comment by douglas-reid** [comment]
:tada: Thanks everyone!

**Pull request merged**